protocols/v2/roles-logic-sv2/src/channel_logic/channel_factory.rs

```diff
@@ -2112,6 +2112,11 @@ impl ProxyExtendedChannelFactory {
             .map(|j| j.0.version)
     }
 
+    /// Get valid jobs
```

Comment:
```suggestion
    /// Return a mining job by its `job_id`.
    /// This can be `None` if we try to fetch a stale job because we clean up the valid jobs array
    /// upon receiving a new `PrevHash`
```